src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1278,9 +1278,27 @@ impl Coordinator {
         plan::DropObjectsPlan {
             drop_ids,
             object_type,
-            referenced_ids: _,
+            referenced_ids,
         }: plan::DropObjectsPlan,
     ) -> Result<ExecuteResponse, AdapterError> {
+        let referenced_ids_hashset = referenced_ids.iter().collect::<HashSet<_>>();
+        let mut objects = Vec::new();
+        for obj_id in &drop_ids {
+            if !referenced_ids_hashset.contains(obj_id) {
+                let sys_obj_id: SystemObjectId = (*obj_id).clone().into();
```

Comment:
1. `obj_id.into()` errs since the `From` trait is implemented to convert `ObjectId`, not `&ObjectId`
2. `(*obj_id).into()` errs since `ObjectId` doesn't implement the `Copy` trait. 

So, I had to use `clone()`. But is there a better way to do this?

Reply:
Would implementing the `Copy` trait be breaking change? Since the enum is already derived with `Clone`, maybe adding the `Copy` trait wouldn't be harmful

Reply:
Functionally, implementing `Copy` would be equivalent to the current code, and I personally prefer being explicit with `Clone`.

Instead you can apply the attached diff which removes the need to `Clone` or `Copy`.

[error-msg.txt](https://github.com/MaterializeInc/materialize/files/13419790/error-msg.txt)